### PR TITLE
feat: Add support for multi file generation from hbs templates

### PR DESCRIPTION
**Pull request by andertun**
Introducing support for multiple files being generated from a single template

If a template e.g. looks like this

```
{#each spec.methods}
//#FILENAME:src/mocks/handlers/{{base}}/{{name}}.ts:create-only
... mock code for {{name}}
{{/each}}
```

It could output multiple files

```
// src/mocks/handlers/messages/getMessages.ts
... mock code for getMessages
```

```
// src/mocks/handlers/messages/addMessage.ts
... mock code for addMessage
```

```
// src/mocks/handlers/messages/deleteMessage.ts
... mock code for deleteMessage
```

```
// src/mocks/handlers/messages/deleteAllMessages.ts
... mock code for deleteAllMessages
```

### CODE REVIEW

File: src/Target.ts [hunk 1]
```diff
@@ -149,39 +149,57 @@ export class Target {
...
+            if (currentMode === 'skip') {
+                return null;
+            }
 
-                if (!mode) {
-                    mode = 'write-always';
-                }
+            // If file content is empty, skip it
+            if (!currentFileContent.some((line) => line.trim())) {
+                return null;
```
Comment (hofmeister) [inline src/Target.ts:166]: looks like this just should return ? (E..g not null just```return;``` )

File: src/Target.ts [hunk 2]
```diff
@@ -149,39 +149,57 @@ export class Target {
...
+        let currentFilename = filename;
+        let currentMode = 'write-always';
+        let currentPermissions = '644';
 
-                if (!permissions) {
-                    permissions = '644';
-                }
+        const processCurrentFile = () => {
+            if (currentMode === 'skip') {
+                return null;
```
Comment (hofmeister) [inline src/Target.ts:161]: looks like this just should return ? (E..g not null just ```return;``` )

File: src/Template.ts [hunk 3]
```diff
@@ -117,12 +117,14 @@ export function create(data: any, context: any, codeFormatter: CodeFormatter): T
 
     handlebarInstance.registerHelper('consumes', function (this: any, type, options: HelperOptions) {
         if (!context.spec.consumers) {
-            return '';
+            return options.inverse(this);
```
Comment (hofmeister) [inline src/Template.ts:120]: What does this do ?

File: src/Target.ts (hunk 2 above)
Comment (andertun) [reply src/Target.ts]: Fixed here fb2fdb49540007f5351dbfe3230a370a0591bcde

File: src/Target.ts (hunk 1 above)
Comment (andertun) [reply src/Target.ts]: Fixed here fb2fdb49540007f5351dbfe3230a370a0591bcde

File: src/Template.ts (hunk 3 above)
Comment (andertun) [reply src/Template.ts:120]: It enables us to use {{else}}

https://handlebarsjs.com/guide/block-helpers.html#conditionals

Comment (hofmeister) [reply src/Template.ts:120]: Nice! Didn't know that one !

**Review by hofmeister — APPROVED** [review]

**Pull request merged**